src/components/Notification/NotificationMessage.tsx

```diff
@@ -0,0 +1,261 @@
+    touchStart.current = e.targetTouches[0].clientX;
+  };
+
+  /**
+   * Touch Move
+   * @param e  TouchEvent
+   */
+  const handleTouchMove = (e: TouchEvent) => {
+    touchEnd.current = e.targetTouches[0].clientX;
+  };
+
+  /**
+   * Touch End
+   */
+  const handleTouchEnd = () => {
+    const threshold = 45;
+    const difference =
+      touchStart.current > touchEnd.current
+        ? touchStart.current - touchEnd.current
+        : touchEnd.current - touchStart.current;
+
+    // If touch distance goes over threshold then close toast
+    if (difference > threshold) {
+      handleClose();
+    }
+  };
+
+  /**
+   * Init interval for countdown
+   * @param time number
+   */
+  const initInterval = (time: number) => {
+    const interval = window.setInterval(() => {
+      const increment = 10;
+
+      if (time > duration) {
+        clearInterval(interval);
+        handleClose();
+      }
+
+      time += increment;
+      // Store ref to start again at that point if need be
+      timeVisible.current = time;
+
+      animateProgress(time, duration);
+    }, 10);
+
+    setIntervalId(interval);
+  };
+
+  /**
+   * Animate progress bar
+   * @param time
+   * @param duration
+   */
+  const animateProgress = (time: number, duration: number) => {
+    progressBar.current?.style.setProperty(
+      '--progress',
+      String(1 - time / duration)
+    );
+  };
+
+  /**
+   * Mouse Enter
+   */
+  const handleMouseEnter = () => {
+    console.log('hit 2');
+
+    if (!pauseOnHover || !autoClose) return;
+    clearInterval(intervalId);
+  };
+
+  /**
+   * Mouse Leave
+   */
+  const handleMouseLeave = () => {
+    if (!pauseOnHover || !autoClose) return;
+    initInterval(timeVisible.current);
+  };
+
+  /**
+   * On Close
+   */
+  const handleClose = useCallback(() => {
+    removeEvents();
+    setIsOpen(false);
+    clearInterval(intervalId);
+  }, []);
+
+  /**
+   * Handle CTA action click
+   */
+  const handleAction = useCallback(() => {
+    typeof callback === 'function' && callback();
+  }, []);
+
+  /**
+   * On fade out complete
+   */
+  const handleOnComplete = useCallback(() => {
+    if (!isOpen) {
+      setIsVisible(false);
+      // dispatch parent to delete me
+      typeof removeSelf === 'function' && removeSelf(id);
+    }
+  }, [isOpen]);
+
+  const fadeAnimation = () => {
+    // Fade to the right
+    let animation = 'translateX(100px)';
+    if (
+      location === NotificationLocationE.BOTTOM_LEFT ||
+      location === NotificationLocationE.TOP_LEFT
+    ) {
+      // Fade to the left
+      animation = 'translateX(-100px)';
+    }
+    return animation;
+  };
+
+  const ProgressBar = (
+    <div
+      ref={progressBar}
+      className={`${styles.progress_bar} ${autoClose && styles.active}`}
+    ></div>
+  );
+
+  return (
+    <FadeIn
+      isVisible={isOpen}
+      onComplete={handleOnComplete}
+      animation={fadeAnimation()}
+    >
+      {isVisible && category === CategoryE.TOAST && (
```

Comment:
So it escalates better in the future maybe I'd move this to something like:
```typescript
const renderMessage = (param: CategoryE) => {
    switch(param) {
      case CategoryE.CRUMB:
        return <CrumbMessage
          description={description}
          ref={notificationRef}
          icon={icon}
          hasIcon={hasIcon}
          mouseEnter={handleMouseEnter}
          mouseLeave={handleMouseLeave}
          close={handleClose}
        />;
      case CategoryE.TOAST:
        return <ToastMessage
          title={title}
          description={description}
          ref={notificationRef}
          callback={callback}
          callbackCta={callbackCta}
          type={type}
          mouseEnter={handleMouseEnter}
          mouseLeave={handleMouseLeave}
          close={handleClose}
          progressBar={ProgressBar}
        />;
    }
```
and leave this as just:
```typescript
{isVisible && renderMessage(category)}
```
(that would make category a mandatory parameter but we probably want that)